hudi-flink-datasource/hudi-flink1.19.x/src/main/java/org/apache/hudi/adapter/MaskingOutputAdapter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.adapter;
+
+import org.apache.flink.streaming.api.operators.Output;
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.LatencyMarker;
+import org.apache.flink.streaming.runtime.streamrecord.RecordAttributes;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.watermarkstatus.WatermarkStatus;
+import org.apache.flink.util.OutputTag;
+
+/** Adapter class for {@code Output} to handle async compaction/clustering service thread safe issues */
+public class MaskingOutputAdapter<OUT> implements Output<StreamRecord<OUT>> {
+
```

Comment:
I just notice that maybe we can use this `MaskingOutputAdapter` to mask out `CollectOutput`.

Reply:
But needs to take an output in constructor. MaskingOutputAdapter(Output<StreamRecord<OUT>> output)

Reply:
`CollectorOutput` is the output?

Reply:
CollectorOutput needs to be removed as it implements the output interface. If we bump the latest flink version to 1.19, the CollectorOutput needs to implement the new method. Then, it can't compile with older flink versions.

Reply:
I feel like if we wrap up the `CollectorOutput` with the `MaskingOutputAdapter`, it should work, masking out the new interfaces is what the `MaskingOutputAdapter` do, that would resolve the compatibility issue.

Reply:
@danny0405 
I am not fully understand the suggestion. Would you please elaborate a little more?

Reply:
Fine, it looks good.